packages/lesmis/src/components/DashboardReport.js

```diff
@@ -29,83 +31,66 @@ const Footer = styled(FlexEnd)`
 `;
 
 export const DashboardReport = React.memo(
-  ({ entityCode, dashboard, year, viewConfig, drillDowns }) => {
-    const { code: itemCode, legacy, periodGranularity, reportCode, name, type } = viewConfig;
-    const { dashboardCode } = dashboard;
-    const { startDate, endDate } = yearToApiDates(year);
+  ({ name, reportCode, startDate, endDate, isEnlarged }) => {
+    const { search } = useLocation();
+    const { entityCode } = useUrlParams();
 
-    const { data, isLoading, isFetching, isError, error } = useDashboardReportData({
+    const { data, isLoading, isFetching, isError, error } = useDashboardReportDataWithConfig({
       entityCode,
       reportCode,
-      dashboardCode,
-      itemCode,
-      legacy,
-      periodGranularity,
       startDate,
       endDate,
     });
 
-    if (type === 'list') {
-      return (
-        <Container>
-          <ListVisual
-            viewContent={{ ...viewConfig, data, startDate, endDate }}
-            isLoading={isLoading}
-            isError={isError}
-            error={error}
-            drillDowns={drillDowns}
-            dashboard={dashboard}
-            DrillDownComponent={DashboardReportModal}
-          />
-        </Container>
-      );
-    }
+    const { reportData, dashboardItemConfig: config, reportCodes } = data;
```

Comment:
`reportCodes` feels like quite a different concern to me, and maybe should be a different query to the one that fetches the data for a specific dashboard item?

Reply:
I know what you mean, it seems like there should be an easier way. All I'm trying to do is to map the dashboardItemCode to the reportCode. I could add another query like useReportCodes or useReportCodeByItemCode that would make another request (probably hit the cache) to the `dashboard` end point. Is that what you mean? Other options I can think of are:
- adding react context somewhere to store the report codes or dashboard data
- iterating on the drilldown configs after they are fetched and adding the reportcodes next to the dashboardItemCodes
- adding the reportCodes to the backend config
- Changing the backend config to be report codes instead of item codes
- Adding the dashboardItemCode to the url

But none of them seem that good to me 🤷‍♂️

Reply:
I was going for using the dashboard item code as the base unit of reference that got passed around, and then within the dashboard report we have another hook that gets the report code at that point. This would ideally hit the cached react query from the `/dashboard` endpoint.

I started this on Tuesday end of day but ended up in circles so gave up. Might have another little crack today if I get a chance